docs: Add documentation for API definition compression (TT-16266) (#1330)
TykTechnologies/tyk-docs: planning-for-production/database-settings.mdx

**Capping Analytics**
Tyk Gateways can generate a lot of analytics data. Be sure to read about [capping your Dashboard analytics](/api-management/tyk-pump#tyk-pump-capping-analytics-data-storage)

**API Definition Compression for MDCB Deployments**

> In [MDCB](/api-management/mdcb) deployments, worker Gateways store API definition backups in Redis to ensure resilience when the control plane is unavailable. For deployments with many or large API definitions, this can consume significant Redis memory.

Reviewer: ```suggestion
In [MDCB](/api-management/mdcb) deployments, Data Plane Gateways store API definition backups in Redis to ensure resilience when the Control Plane is unavailable. For deployments with many or large API definitions, this can consume significant Redis memory.
```

We should align the documentation with Control Plane / Data Plane terminology and stop using "worker Gateway".

> You can enable Zstandard compression to reduce the storage footprint by setting [`storage.compress_api_definitions`](/tyk-oss-gateway/configuration#storage-compress_api_definitions) to `true` in your worker Gateway configuration:

Reviewer: ```suggestion
You can enable Zstandard compression to reduce the storage footprint by setting [`storage.compress_api_definitions`](/tyk-oss-gateway/configuration#storage-compress_api_definitions) to `true` in your Data Plane Gateway configuration:
```

```json
"storage": {
  ...
  "compress_api_definitions": true
}
```

> 

Reviewer: Taken from the godocs comment in `config.go`:

```suggestion
The Gateway will seamlessly read uncompressed API definitions if they are already stored in Redis in that format. When the Data Plane is reloaded, the definitions will be compressed before being stored.

Tyk can support and compress API definitions up to 100 MB in size. Typically you should expect to see approximately 75% compression for Tyk OAS APIs.
```

**Capping Analytics**
Tyk Gateways can generate a lot of analytics data. Be sure to read about [capping your Dashboard analytics](/api-management/tyk-pump#tyk-pump-capping-analytics-data-storage)

**API Definition Compression for MDCB Deployments**

In [MDCB](/api-management/mdcb) deployments, worker Gateways store API definition backups in Redis to ensure resilience when the control plane is unavailable. For deployments with many or large API definitions, this can consume significant Redis memory.

> You can enable Zstandard compression to reduce the storage footprint by setting [`storage.compress_api_definitions`](/tyk-oss-gateway/configuration#storagecompress_api_definitions) to `true` in your worker Gateway configuration:

Reviewer 2: ```suggestion
You can enable Zstandard compression to reduce the storage footprint by setting [`storage.compress_api_definitions`](/tyk-oss-gateway/configuration#storage-compress_api_definitions) to `true` in your worker Gateway configuration:
```

```json
"storage": {
  ...
  "compress_api_definitions": true
}
```

This feature:
- Compresses API definitions before storing them in Redis backups
- Automatically decompresses when loading backups
- Is backward compatible - compressed and uncompressed data are handled transparently

> - Can significantly reduce Redis memory usage depending on your API definitions

Reviewer 2: ```suggestion
```
